py-polars/tests/unit/operations/map/test_inefficient_map_warning.py

```diff
@@ -254,7 +272,11 @@ def test_parse_apply_functions(col: str, func: str, expr_repr: str) -> None:
             x=pl.col(col),
             y=pl.col(col).map_elements(eval(func)),
         )
-        assert_frame_equal(result_frame, expected_frame)
+        assert_frame_equal(
+            result_frame,
+            expected_frame,
+            check_dtype=(".dt." not in suggested_expression),
```

Comment:
not opposed to this, but I guess an alternative could be to add `.cast` at the end of the suggestion, like is done for the bitwise shifts?

Reply:
We don't want the suggestion to `cast` though; in this case we want them to be using the optimal (smaller) types  ;)